opensearch/src/main/java/org/opensearch/sql/opensearch/data/type/OpenSearchDataType.java

```diff
@@ -6,139 +6,254 @@
 
 package org.opensearch.sql.opensearch.data.type;
 
-import static org.opensearch.sql.data.type.ExprCoreType.STRING;
-import static org.opensearch.sql.data.type.ExprCoreType.UNKNOWN;
-
-import com.google.common.collect.BiMap;
-import com.google.common.collect.ImmutableBiMap;
 import com.google.common.collect.ImmutableMap;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.List;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
 import java.util.Map;
-import lombok.RequiredArgsConstructor;
+import java.util.function.BiConsumer;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
 import org.opensearch.sql.data.type.ExprCoreType;
 import org.opensearch.sql.data.type.ExprType;
 
 /**
- * The extension of ExprType in Elasticsearch.
+ * The extension of ExprType in OpenSearch.
  */
-@RequiredArgsConstructor
-public enum OpenSearchDataType implements ExprType {
-  /**
-   * OpenSearch Text. Rather than cast text to other types (STRING), leave it alone to prevent
-   * cast_to_string(OPENSEARCH_TEXT).
-   * Ref: https://www.elastic.co/guide/en/elasticsearch/reference/current/text.html
-   */
-  OPENSEARCH_TEXT(Collections.singletonList(STRING), "string") {
-    @Override
-    public boolean shouldCast(ExprType other) {
-      return false;
-    }
-  },
+@EqualsAndHashCode
+public class OpenSearchDataType implements ExprType, Serializable {
 
   /**
-   * OpenSearch multi-fields which has text and keyword.
-   * Ref: https://www.elastic.co/guide/en/elasticsearch/reference/current/multi-fields.html
+   * The mapping (OpenSearch engine) type.
    */
-  OPENSEARCH_TEXT_KEYWORD(Arrays.asList(STRING, OPENSEARCH_TEXT), "string") {
-    @Override
-    public boolean shouldCast(ExprType other) {
-      return false;
-    }
-  },
+  public enum MappingType {
+    Invalid(null, ExprCoreType.UNKNOWN),
+    Text("text", ExprCoreType.UNKNOWN),
+    Keyword("keyword", ExprCoreType.STRING),
+    Ip("ip", ExprCoreType.UNKNOWN),
+    GeoPoint("geo_point", ExprCoreType.UNKNOWN),
```

Comment:
question? does geo_point will in response schema? if yes, does it means?
1. geo_point type should be supported in core.
2. jdbc driver should support geo_point

Reply:
Query `select * from datatypes` (a table from IT) response in legacy:
```json
{
  "schema": [
    {
      "name": "boolean_value",
      "type": "boolean"
    },
    {
      "name": "double_range_value",
      "type": "double_range"
    },
    {
      "name": "ip_value",
      "type": "ip"
    },
    {
      "name": "long_range_value",
      "type": "long_range"
    },
    {
      "name": "keyword_value",
      "type": "keyword"
    },
    {
      "name": "date_range_value",
      "type": "date_range"
    },
    {
      "name": "binary_value",
      "type": "binary"
    },
    {
      "name": "date_value",
      "type": "date"
    },
    {
      "name": "text_value",
      "type": "text"
    },
    {
      "name": "nested_value",
      "alias": "",
      "type": "text"
    },
    {
      "name": "object_value",
      "alias": "",
      "type": "text"
    }
  ],
  "total": 1,
  "datarows": [[
    true,
    {
      "lte": 4,
      "gte": 1
    },
    "127.0.0.1",
    {
      "lte": 100500,
      "gte": 0
    },
    "keyword",
    {
      "lte": "2019-05-15",
      "gte": "2019-05-01"
    },
    "U29tZSBiaW5hcnkgYmxvYg==",
    "2020-10-13 13:00:00.000",
    "text",
    [
      {
        "last": "Smith",
        "first": "John"
      },
      {
        "last": "White",
        "first": "Alice"
      }
    ],
    {
      "last": "Dale",
      "first": "Dale"
    }
  ]],
  "size": 1,
  "status": 200
}
```
From V2 without these changes:
```json
{
  "schema": [
    {
      "name": "date_value",
      "type": "timestamp"
    },
    {
      "name": "boolean_value",
      "type": "boolean"
    },
    {
      "name": "ip_value",
      "type": "ip"
    },
    {
      "name": "text_value",
      "type": "text"
    },
    {
      "name": "nested_value",
      "type": "nested"
    },
    {
      "name": "object_value",
      "type": "object"
    },
    {
      "name": "keyword_value",
      "type": "keyword"
    },
    {
      "name": "binary_value",
      "type": "binary"
    },
    {
      "name": "geo_value",
      "type": "geo_point"
    }
  ],
  "datarows": [
    [
      "2020-10-13 13:00:00",
      true,
      "127.0.0.1",
      "text",
      [
        {
          "last": "Smith",
          "first": "John"
        },
        {
          "last": "White",
          "first": "Alice"
        }
      ],
      {
        "last": "Dale",
        "first": "Dale"
      },
      "keyword",
      "U29tZSBiaW5hcnkgYmxvYg==",
      {
        "lat": 40.71,
        "lon": 74.0
      }
    ]
  ],
  "total": 1,
  "size": 1,
  "status": 200
}
```
And with these changes:
```json
{
  "schema": [
    {
      "name": "text_value",
      "type": "text"
    },
    {
      "name": "date_value",
      "type": "timestamp"
    },
    {
      "name": "boolean_value",
      "type": "boolean"
    },
    {
      "name": "ip_value",
      "type": "ip"
    },
    {
      "name": "nested_value",
      "type": "nested"
    },
    {
      "name": "object_value",
      "type": "object"
    },
    {
      "name": "keyword_value",
      "type": "keyword"
    },
    {
      "name": "binary_value",
      "type": "binary"
    },
    {
      "name": "geo_value",
      "type": "geo_point"
    }
  ],
  "datarows": [
    [
      "text",
      "2020-10-13 13:00:00",
      true,
      "127.0.0.1",
      [
        {
          "last": "Smith",
          "first": "John"
        },
        {
          "last": "White",
          "first": "Alice"
        }
      ],
      {
        "last": "Dale",
        "first": "Dale"
      },
      "keyword",
      "U29tZSBiaW5hcnkgYmxvYg==",
      {
        "lat": 40.71,
        "lon": 74.0
      }
    ]
  ],
  "total": 1,
  "size": 1,
  "status": 200
}
```

And JDBC doesn't support few types including `geo_point`:
![image](https://user-images.githubusercontent.com/88679692/220450623-9e4d126c-41c8-4852-9363-1fd857fd8184.png)